atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/BackgroundSweeperImpl.java

```diff
@@ -208,26 +231,25 @@ boolean runOnce() {
                 });
     }
 
-    /**
-     * Check whether the table being swept was dropped. If so, stop sweeping it and move on.
-     *
-     * @return Whether the table being swept was dropped
-     */
-    boolean checkAndRepairTableDrop() {
+    private SweepOutcome determineCauseOfFailure(Exception originalException) {
         try {
             Set<TableReference> tables = specificTableSweeper.getKvs().getAllTableNames();
             Optional<SweepProgress> progress = specificTableSweeper.getTxManager().runTaskReadOnly(
                     specificTableSweeper.getSweepProgressStore()::loadProgress);
+
             if (!progress.isPresent() || tables.contains(progress.get().tableRef())) {
-                return false;
+                log.warn("The background sweep job failed unexpectedly; will retry with a lower batch size...",
+                        originalException);
+                return SweepOutcome.RETRYING_WITH_SMALLER_BATCH;
             } else {
                 specificTableSweeper.getSweepProgressStore().clearProgress();
-                return true;
+                log.info("The table being swept by the background sweeper was dropped, moving on...");
```

Comment:
How are we sure the table is dropped when sweeping?

Maybe the `progress` table just failed to updateand hence it is not null but contains the update from the previous iteration. 
Shouldnt we check `tables.contains(tableToSweep.get())) == false` here? 

I know this was the behavior in the past but now that we are collecting these outcomes and alerting based on these, we should be sure that the outcome recorded is true.

Reply:
Yeah, I as trying to avoid changing behaviour here but it does look like a bit of leap to say the table must have been dropped here.

I could change the logic to:
```
if(progress.isPresent && !tables.contains(progress.getTable)) {
    return TABLE_DROPPED_WHILE_SWEEPING;
} else {
    return RETRYING_WITH_SMALLER_BATCH;
}
```

At which point I might remove the `RETRYING_WITH_SMALLER_BATCH` outcome and just `ERROR` - we end up retrying with a smaller batch anyway.

Thoughts?